pkgs/desktops/lomiri/applications/lomiri/default.nix

```diff
@@ -0,0 +1,288 @@
+      url = "https://salsa.debian.org/ubports-team/lomiri/-/raw/ebbe0f3f568bd145bb58a2e47f7112442328a0a5/debian/patches/2004_qt-qpa-platform-is-wayland.patch";
+      hash = "sha256-4C6X2TW+yjZhqYPIcQ3GJeTKbz785i7p/DpT+vX1DSQ=";
+    })
+    (fetchpatch {
+      name = "1002-lomiri-Fix-Lomiri-greeter.patch";
+      url = "https://salsa.debian.org/ubports-team/lomiri/-/raw/ebbe0f3f568bd145bb58a2e47f7112442328a0a5/debian/patches/1008_lomiri-greeter-wayland.patch";
+      excludes = [ "data/lomiri-greeter.desktop.in.in" ]; # conflict with GNUInstallDirs patch
+      hash = "sha256-XSSxf06Su8PMoqYwqevN034b/li8G/cNXjrqOXyhTRg=";
+    })
+    (fetchpatch {
+      name = "1003-lomiri-Hide-launcher-in-greeter-mode.patch";
+      url = "https://salsa.debian.org/ubports-team/lomiri/-/raw/ebbe0f3f568bd145bb58a2e47f7112442328a0a5/debian/patches/0002_qml-shell-hide-and-disallow-launcher-in-greeter-only-mode.patch";
+      hash = "sha256-R0aMlb7N7XACCthML4SQSd0LvbadADfdQJqrYFhmujk=";
+    })
+    (fetchpatch {
+      name = "1004-lomiri-Dont-reset-OSK-setting.patch";
+      url = "https://salsa.debian.org/ubports-team/lomiri/-/raw/ebbe0f3f568bd145bb58a2e47f7112442328a0a5/debian/patches/2005_dont-reset-alwaysShowOsk-to-system-defaults-on-login.patch";
+      hash = "sha256-guq/Ykcq4WcuXxNKO1eA4sJFyGSpZo0gtyFTdeK/GeE=";
+    })
+
+    ./9901-lomiri-Disable-Wizard.patch
+  ];
+
+  postPatch = ''
+    # Part of greeter fix, applies separately due to merge conflicts
+    substituteInPlace data/lomiri-greeter.desktop.in.in \
+      --replace-fail '@CMAKE_INSTALL_FULL_BINDIR@/lomiri-greeter-wrapper @CMAKE_INSTALL_FULL_BINDIR@/lomiri --mode=greeter' '@CMAKE_INSTALL_FULL_BINDIR@/lomiri --mode=greeter' \
+      --replace-fail 'X-LightDM-Session-Type=mir' 'X-LightDM-Session-Type=wayland'
+
+    # Need to replace prefix
+    substituteInPlace data/systemd-user/CMakeLists.txt \
+      --replace-fail 'pkg_get_variable(SYSTEMD_USERUNITDIR systemd systemduserunitdir)' 'pkg_get_variable(SYSTEMD_USERUNITDIR systemd systemduserunitdir DEFINE_VARIABLES prefix=''${CMAKE_INSTALL_PREFIX})'
+
+    # Don't embed full paths into regular desktop files (but do embed them into lightdm greeter one)
+    substituteInPlace data/{indicators-client,lomiri}.desktop.in.in \
+      --replace-fail '@CMAKE_INSTALL_FULL_BINDIR@/' ""
+
+    # Exclude tests that don't compile (Mir headers these relied on were removed in mir 2.9)
+    # fatal error: mirtest/mir/test/doubles/stub_surface.h: No such file or directory
+    substituteInPlace tests/mocks/CMakeLists.txt \
+      --replace-fail 'add_subdirectory(QtMir/Application)' ""
+
+    #substituteInPlace plugins/AccountsService/CMakeLists.txt \
+    #  --replace-fail 'CMAKE_INSTALL_DATADIR' 'CMAKE_INSTALL_FULL_DATADIR'
+
+    # NixOS-ify
+
+    # Use Nix flake instead of Canonical's Ubuntu logo
+    rm qml/Launcher/graphics/home.svg
+    ln -s ${./lomiri-launcher-logo.svg} qml/Launcher/graphics/home.svg
```

Comment:
When I last fiddled with this ~ a year ago, using regular snowflake svg didn't work correctly / looked broken in this context. I'll take another stab at it.

Reply:
Ah right. Using `${nixos-icons}/share/icons/hicolor/scalable/apps/nix-snowflake-white.svg` *works*, but the icon is displayed abit squished.

![image](https://github.com/NixOS/nixpkgs/assets/23431373/2b79c950-9a6c-4a7e-97f8-bb0c9c0b92f2)

This is because none of the snowflake graphics have correct dimensions. They should all have widths that are equal to their heights & the XDG icon directories they're in, but none of them have that.

https://github.com/NixOS/nixos-artwork/issues/42

```
/nix/store/9v1j72wdckazf1301lag60kapx0zcxcv-nixos-icons-2021-02-24/share/icons/hicolor/32x32/apps/nix-snowflake-white.png PNG 32x28 32x28+0+0 8-bit Grayscale Gray 929B 0.000u 0:00.000
/nix/store/9v1j72wdckazf1301lag60kapx0zcxcv-nixos-icons-2021-02-24/share/icons/hicolor/128x128/apps/nix-snowflake-white.png PNG 128x111 128x111+0+0 8-bit Grayscale Gray 4159B 0.000u 0:00.000
/nix/store/9v1j72wdckazf1301lag60kapx0zcxcv-nixos-icons-2021-02-24/share/icons/hicolor/scalable/apps/nix-snowflake-white.svg SVG 535x464 535x464+0+0 16-bit sRGB 6695B 0.000u 0:00.007
```

I guess having it squished would be fine, but it just really bothered me to look at that all the time…

Reply:
Fixing this upstream: https://github.com/NixOS/nixos-artwork/pull/118

I guess we'll use the not-quite-square ones until then, so the fix propagates into here eventually…

Reply:
Fixed upstream, updating `nixos-icons` in https://github.com/NixOS/nixpkgs/pull/303068 . Will switch now, and let that propagate via the update.